src/System.Private.CoreLib/shared/System/Threading/AsyncLocal.cs

```diff
@@ -56,12 +57,13 @@ public AsyncLocal(Action<AsyncLocalValueChangedArgs<T>>? valueChangedHandler)
             m_valueChangedHandler = valueChangedHandler;
         }
 
+        [MaybeNull]
```

Comment:
Does this apply to the getter as well? Shouldn't we add a `[return: MaybeNull]` here as well?

Reply:
Ah it will be implicit by the type used when creating an instance of this class.

Reply:
> Does this apply to the getter as well?

It only applies to the getter.  `MaybeNull` is only for the out direction, so it's not applicable to the setter.  If this were `AllowNull`, then it would only be relevant to the setter.